### PR TITLE
getEvents performance improvements

### PR DESCRIPTION
**Pull request by kyscott18**

### CODE REVIEW

File: packages/core/src/common/options.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ export type Options = {
   indexingCacheBytes: number;
 
   syncMaxIntervals: number;
+  syncEventsQuerySize: number;
```
Comment (0xOlias) [inline packages/core/src/common/options.ts:38]: Naming ideas:
```
indexingCacheMaxBytes
indexingCacheFlushRatio

syncStoreMaxDistinctIntervals
syncStoreMaxBatchSize
```

File: packages/core/src/sync-store/postgres/store.ts [hunk 2]
```diff
@@ -2202,114 +2215,35 @@ export class PostgresSyncStore implements SyncStore {
...
       });
 
-      const hasNextPage = events.length === limit + 1;
-
-      if (!hasNextPage) {
-        yield events;
-        break;
+      // set fromCursor + seconds
+      if (events.length === 0) {
+        this.seconds = Math.round(this.seconds * 2);
```
Comment (0xOlias) [inline packages/core/src/sync-store/postgres/store.ts:2220]: Could consider making these constants explicit options, but fine to skip for now I think.

File: packages/core/src/sync-store/sqlite/store.ts [hunk 3]
```diff
@@ -2059,10 +2072,10 @@ export class SqliteSyncStore implements SyncStore {
                 "transactionReceipts.type as txr_type",
               ]),
           )
-          .where("events.checkpoint", ">", cursor)
-          .where("events.checkpoint", "<=", encodedToCheckpoint)
+          .where("events.checkpoint", ">", fromCursor)
+          .where("events.checkpoint", "<=", toCursor)
```
Comment (0xOlias) [inline packages/core/src/sync-store/sqlite/store.ts:2076]: Worth double-checking the exclusivity rules here again (I think we're good though).

File: packages/core/src/sync-store/store.test.ts [hunk 4]
```diff
@@ -2661,7 +2662,7 @@ test("getEvents filters on toCheckpoint (inclusive)", async (context) => {
 
   const ag = syncStore.getEvents({
     sources: [sources[0]],
-    fromCheckpoint: zeroCheckpoint,
+    fromCheckpoint: createBlockCheckpoint(rpcData.block2.block, false),
```
Comment (0xOlias) [inline packages/core/src/sync-store/store.test.ts:2665]: For my understanding: Now, it's no longer OK/safe to pass zeroCheckpoint as fromCheckpoint?

File: packages/core/src/sync/service.ts [hunk 5]
```diff
@@ -30,6 +32,13 @@ export type Service = {
 
   // state
   checkpoint: Checkpoint;
+  /** Checkpoint of the earliest start block. */
```
Comment (0xOlias) [inline packages/core/src/sync/service.ts:35]: Great comments here

File: packages/core/src/sync/service.ts [hunk 6]
```diff
@@ -279,6 +316,20 @@ export const create = async ({
           sources: networkSources,
           requestQueue,
           cachedTransport: cachedTransport({ requestQueue, syncStore }),
+          startCheckpoint: {
```
Comment (0xOlias) [inline packages/core/src/sync/service.ts:319]: Could consider evaluating these two above, like initialFinalizedCheckpoint

**Review by 0xOlias — APPROVED** [review]
All nits - looks great

File: packages/core/src/sync-store/postgres/store.ts [hunk 7]
```diff
@@ -67,12 +67,16 @@ export class PostgresSyncStore implements SyncStore {
...
+  private seconds: number;
+
   constructor({
     db,
     common,
   }: { db: HeadlessKysely<SyncStoreTables>; common: Common }) {
     this.db = db;
     this.common = common;
+
+    this.seconds = common.options.syncEventsQuerySize * 2;
```
[the remark this answers is not in the document]
Comment (kyscott18) [reply packages/core/src/sync-store/postgres/store.ts:79]: It's helpful to be an instance variable for the "cold" cases, when sync is happening at the same time as indexing

File: packages/core/src/sync-store/store.test.ts (hunk 4 above)
Comment (kyscott18) [reply packages/core/src/sync-store/store.test.ts:2665]: it is fine, just does a ton of extra queries

**Pull request merged**